builtin/credential/aws/path_role_test.go

```diff
@@ -986,6 +988,87 @@ func TestAwsVersion(t *testing.T) {
 	}
 }
 
+// This test was used to reproduce https://github.com/hashicorp/vault/issues/7418
+// and verify its fix.
+// Please run it at least 3 times to ensure that passing tests are due to actually
+// passing, rather than the region being randomly chosen tying to the one in the
+// test through luck.
+func TestRoleResolutionWithSTSEndpointConfigured(t *testing.T) {
+	if enabled := os.Getenv("VAULT_ACC"); enabled == "" {
```

Comment:
helper/testhelpers/logical/testing.go defines a `TestEnvVar` variable as this; use that instead? Or is that being deprecated?

Reply:
We could use that as long as it doesn't create a circular dependency. The beginning of this test is a copy paste of https://github.com/hashicorp/vault/blob/805a0bc9b4605aec2b1bdf5aa1cff5cc43dc78f6/physical/s3/s3_test.go#L29-L43.

Reply:
Wow, neat that GH shows the code inline now! :-)

Reply:
Resolved in https://github.com/hashicorp/vault/pull/7922/commits/13dfdc12974a2aec8268d5579361f8b4baf6cf38